servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ServiceFactoryCustomizationTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.netty;
+
+import io.servicetalk.grpc.api.GrpcExecutionStrategies;
+import io.servicetalk.grpc.api.GrpcServiceContext;
+import io.servicetalk.router.api.RouteExecutionStrategy;
+
+import io.grpc.examples.helloworld.Greeter;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class ServiceFactoryCustomizationTest {
+
+    @Test
+    void usesCustomRouteExecutionStrategyFactory() throws Exception {
+        final AtomicBoolean customFactoryCalled = new AtomicBoolean(false);
+
+        GrpcServers
+                .forAddress(localAddress(0))
+                .listenAndAwait(new Greeter.ServiceFactory.Builder()
+                        .routeExecutionStrategyFactory(id -> {
```

Comment:
Instead of creating a new test class, consider modifying the existing `ExecutionStrategyTest`. After your changes, its `RouteExecutionStrategy` uses deprecated ctors. It will be great to migrate those to new API.
However, it's also great to make sure that deprecated API still works as expected. So, instead of just migrating to the new API, consider also keeping the deprecated variant for now.